src/examples/basic/form.test.jsx

```diff
@@ -0,0 +1,199 @@
+    // defines the contract of the component
+    // internal state (component state/or data generated from utils in the component) are behavioral not contractual
+    // state pulled off of redux via useSelector is
+    // When data is expected to be passed into a component we need to define that contract and test for contract violations
+    // You do not need to split out contract tests from behavioral tests
+    // I'm splitting them out as an exercise
+    // You can think of Component contract tests as looking for your 200, 400, and 500 cases
+
+    // Food for thought: Is it okay to explicitly throw errors or allow errors to bubble up?
+    // Should components just be written to handle all edge cases gracefully?
+    // Answer: It depends, sometimes handling errors gracefully means failing silently
+    // For non-essential components that's fine, but if the entire form were to fail to render
+    // and an error wasn't logged or thrown, that results in a loss of observability in the client
+    // sometimes throwing custom excepts makes it easier for devs to debug later
+
+    // Honestly if we have a statically typed language this wouldn't even be a problem
+    test('BasicForm renders a form with a header when the required header is passed in', () => {
+      const props = {
+        nonOptionalHeader: 'PB & J Anonymous',
+      };
+      const { container } = renderComponent(props);
+      // This is essentially a renders without crashing test
+      expect(container.firstChild).not.toBeEmpty();
+    });
+    test('BasicForm throws an error if nonOptionalHeader is falsy', () => {
+      const props = {
+        nonOptionalHeader: undefined,
+      };
+      // You can pass expect a callback function to help test for expected errors
+      expect(() => renderComponent(props)).toThrowError('nonOptionalHeader is non-optional');
+    });
+    test('BasicForm throws an error if nonOptionalHeader isn\'t a string', () => {
+      const props = {
+        nonOptionalHeader: 21122,
+      };
+      // You can pass expect a callback function to help test for expected errors
+      expect(() => renderComponent(props)).toThrow();
+    });
+  });
+
+  // Behavior testing - the tests you're used to writing
+  describe('Behavior Tests', () => {
+    // Component behavior:
+    // First render
+    // Rerenders - UI interaction or prop changes (bulk of testing)
+    // Unmount
+    test('renders an interactive login form with empty values', () => {
+      const props = {
+        nonOptionalHeader: 'PB & J Anonymous',
+      };
+      const {
+        getByText,
+        getByPlaceholderText,
+        getByLabelText,
+      } = renderComponent(props);
+
+      // Basic check that an element is visible
+      const formTitle = getByText(props.nonOptionalHeader, { exact: false });
+      expect(formTitle).toBeVisible();
+
+      // Slightly better selector than getByText since we tend to hide
```

Comment:
Worth nothing, that instructions in placeholders is generally a no no, usability/accessibility wise. As soon as someone types in an input box, they lose the instructions.

Reply:
yeah, I believe they have this in their docs too. I'll change it to something else